### DISCUSSION TITLE
Does Mockoon support the soap protocol?

### DISCUSSION THREAD

**Discussion by heyyuyu** [mockoon/mockoon#1360] [Help]
SOAP protocol request body is XML，and the xml contains method like this:
```
<?xml version="1.0" encoding="UTF-8"?>
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
<soapenv:Header/>
<soapenv:Body>
<bcs:QueryOfferingRentCycleRequestMsg> ....
```

How to get the method 'QueryOfferingRentCycleRequestMsg' in the rule?
 does Mockoon support it?Thanks!

**Comment by 255kb** [comment]
There is no SOAP support, but XML get parsed and translated to JSON behind the scene. You can get a preview of how it gets parsed using [this tool](https://mockoon.com/tools/xml-to-json/). 
You may be able to fetch the name of the property using JSONPath either by using the `bodyRaw` helper in the body, or in a rule: 

In a template: 

```
{{bodyRaw '$.`soapenv:Envelope.`soapenv:Body.*~'}}
```

With a rule: 
![image](https://github.com/mockoon/mockoon/assets/7489814/c7b7764c-0f96-4f63-ad49-f0bfceb71773)

**Reply by heyyuyu** [reply]
it seems not working,I use {{bodyRaw '$.`soapenv:Envelope.`soapenv:Body.*~'}} but return empty:
![image](https://github.com/mockoon/mockoon/assets/19744761/7883901a-3e12-43fc-90b2-554ec520737b)

![image](https://github.com/mockoon/mockoon/assets/19744761/f00f6435-ae31-481c-8986-7a831fbf5fa0)

**Reply by 255kb** [reply]
Did you send a `Content-Type: text/xml` header with your request?

I tried with the following template and XML request and it worked:

```
{{#if (eq (bodyRaw '$.`soapenv:Envelope.`soapenv:Body.*~') 'ser:getMo')}}
getMo
{{/if}}
```

Request:
```xml
<soapenv:Envelope>
	<soapenv:Body>
		<ser:getMo>
			<userName>?</userName>
			<password>?</password>
		</ser:getMo>
	</soapenv:Body>
</soapenv:Envelope>
```

**Reply by heyyuyu** [reply]
my content type is `Content-type: text/xml;charset=UTF-8`
![image](https://github.com/mockoon/mockoon/assets/19744761/b3f6adb5-829c-4c60-ba00-94466c3c4e2f)

**Reply by 255kb** [reply]
I tested with the same setup and it worked.I don't see what is wrong. 
Could you maybe share the full mockoon setup so I can try to reproduce the issue? 
![image](https://github.com/mockoon/mockoon/assets/7489814/87f9278a-27f7-4ffa-99d7-f40192334417)

**Reply by heyyuyu** [reply]
{
    "uuid": "5149b4c9-8edc-4edb-a391-8cd888808a42",
    "type": "http",
    "documentation": "",
    "method": "post",
    "endpoint": "test",
    "responses": [
        {
            "uuid": "f9365757-bb92-4a4b-9a05-b0d0bf32414d",
            "body": "{{#if (eq (bodyRaw '$.`soapenv:Envelope.`soapenv:Body.*~') 'ser:getMo')}}\r\ngetMo\r\n{{/if}}",
            "latency": 0,
            "statusCode": 200,
            "label": "",
            "headers": [],
            "bodyType": "INLINE",
            "filePath": "",
            "databucketID": "",
            "sendFileAsBody": false,
            "rules": [],
            "rulesOperator": "OR",
            "disableTemplating": false,
            "fallbackTo404": false,
            "default": true,
            "crudKey": "id"
        }
    ],
    "enabled": true,
    "responseMode": null
}

**Reply by 255kb** [reply]
Same for me, it's working. Be sure you didn't disable the templating in the response settings: 
![image](https://github.com/mockoon/mockoon/assets/7489814/b0370cf9-ddb0-4914-a222-1c02fab1bca1)

**Reply by heyyuyu** [reply]
It worked,it's solved when I upgraded to latest version,sorry for bothering you,thanks again

**Reply by 255kb** [reply]
Great!
